### PR TITLE
Adding resiliency in RedisClusterClient

### PR DESCRIPTION
**Pull request by andrijaperovic**
- Handles TCP keepidle in failover scenario with Redis for stateful connection
- Adding auth with SSL support for REDIS_CLUSTER type configuration
- Handle cluster details endpoint of Redis Cluster returning IP in REDIS_CLUSTER type connection

@snowmanmsft @adchia

### CODE REVIEW

File: storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java [hunk 1]
```diff
@@ -62,18 +71,81 @@ private RedisClusterClient(Builder builder) {
...
+        ClientResources.builder()
+            .nettyCustomizer(
+                new NettyCustomizer() {
+                  @Override
+                  public void afterBootstrapInitialized(Bootstrap bootstrap) {
+                    bootstrap.option(EpollChannelOption.TCP_KEEPIDLE, 15);
+                    bootstrap.option(EpollChannelOption.TCP_KEEPINTVL, 5);
+                    bootstrap.option(EpollChannelOption.TCP_KEEPCNT, 3);
+                    // Socket Timeout (milliseconds)
+                    bootstrap.option(EpollChannelOption.TCP_USER_TIMEOUT, 60000);
```
Comment (adchia) [inline storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java:118]: thoughts on having these be exposed in the cluster config?

Comment (andrijaperovic) [reply storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java:118]: I think that makes sense. Also, these are very cloud provider specific for Azure Cache for Redis in terms of how socket timeouts are handled (https://github.com/lettuce-io/lettuce-core/issues/1428), so I'm wondering if it makes sense to subclass `RedisClusterClient` and make it cloud-provider specific (e.g. `AzureRedisClusterClient`, `AwsRedisClusterClient`, `GcpRedisClusterClient`, etc.).

Comment (andrijaperovic) [reply storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java:118]: Cloud provider can be set via config property (e.g. `cloud_provider: azure`).

Comment (andrijaperovic) [reply storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java:118]: Do you have a usecase for a stateful redis connection in newer versions of feast?

**Review by adchia — APPROVED** [review]
/lgtm

File: storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java (hunk 1 above)
Comment (adchia) [reply storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java:118]: i.e. this wouldnt hold true if you wanted to run your own Redis cluster in Azure?

Might make sense then to instead allow for additional azure_configs within the RedisClusterConfig?

**Pull request merged**